tools/ci/actions/deepen-fetch.sh

```diff
@@ -41,7 +41,10 @@ else
   shift 5
 fi
 
-FETCH_HEAD=$(git rev-parse FETCH_HEAD)
+# Record FETCH_HEAD (if it hasn't been already)
```

Comment:
Maybe comment on when this can happen in practice? (It's a bit mysterious to me.)

Reply:
Oh yes, the commit message explains why but not the comment! I changed it to "(if it hasn't been by a previous step)" which I think is clearer?